apps/survey/src/components/elements/CategoryContentsView.vue

```diff
@@ -129,10 +138,46 @@ export default defineComponent({
   methods: {
     categorySelected(category: CategoryHeader): void {
       this.$emit('category-selected', category);
+      const params: GtmEventParams = {
+        food_category: category.name,
+        search_term: this.searchTerm,
+        search_count: this.searchCount,
+        percent_scrolled: this.percentScrolled,
+        noninteraction: false,
+      };
+      for (const key in GtmEvents) {
```

Comment:
this looping is unnecessary, types you establish already guard that so you can just assign it:
```
params.event = 'selectFoodCategory;
....
params.scheme_prompts = 'foods'
```